src/mysky.ts

```diff
@@ -345,31 +685,67 @@ export class MySky {
         return;
       }
 
-      // Parse the seed.
-      const seed = new Uint8Array(JSON.parse(newValue));
-
-      // Launch the new permissions provider.
-      this.permissionsProvider = launchPermissionsProvider(seed);
+      try {
+        // Parse the seed.
+        const seed = new Uint8Array(JSON.parse(newValue));
 
-      // If the email is found, then set up auto-login on Main MySky.
-      const email = localStorage.getItem(EMAIL_STORAGE_KEY);
-      if (email) {
-        // Clear the stored email.
+        // Connect to siasky.net first.
         //
-        // The email can be cleared here because `localStorage` is only used to
-        // marshal the email from MySky UI over to the invisible MySky iframe.
-        // We don't clear the seed because we need it in storage so that users
-        // are automatically logged-in, when possible. But for the email, it
-        // should be stored on MySky, as the local storage can get cleared,
-        // users can move across browsers etc.
-        localStorage.removeItem(EMAIL_STORAGE_KEY);
-
-        // TODO: Set up auto-login.
+        // NOTE: Don't use the stored preferred portal here because we are just
+        // logging into a new account and need to get the user settings for the
+        // first time. Always use siasky.net.
+        this.client = getLoginClient(seed, null);
+
+        // Login to portal.
+        {
```

Comment:
Yes, it's a block scope.

Reply:
Was this confusing to you? I googled to see if it was standard and some people on StackOverflow were confused by it. Not sure how seriously to take that, though, because people on the internet have different experience levels. To my eyes this looks pretty clear. I wanted to avoid extracting this into a new function because having a lot of nested functions can also be hard to follow.